packages/frontend/src/components/EditorLayout/index.tsx

```diff
@@ -124,6 +109,13 @@ export default function EditorLayout() {
     return <InvalidEditorPage />
   }
 
+  // for loading demo video
+  const showDemo = searchParams.get('showDemo')
+  const flowTemplate = flow?.template
+  const isDemoTemplate = flowTemplate?.tags?.some((tag) => tag === 'demo')
```

Comment:
why dont u check for existence of flow.template.demoVideoDetails instead? there's no need to check tags as well right

Reply:
can move this block after 
if (!flow) check so u dont need the optional chaining